test/SlowTests/Issues/RavenDB-15900.cs

```diff
@@ -167,90 +167,103 @@ public override void SetResponse(JsonOperationContext context, BlittableJsonRead
-                await AssertRaftIndexToBeUpdatedOnNodesAsync(leaderLastIndex, nodes);
+            foreach (var server in Servers)
+            {
+                Assert.False(server.ServerStore.DatabasesLandlord.IsDatabaseLoaded(database));
+            }
 
-                var cmd2 = new CreateDatabaseOperation.CreateDatabaseCommand(new DatabaseRecord("Toli"), 1);
+            foreach (var server in Servers)
+            {
+                var res = await WaitForValueAsync(() => server.ServerStore.Engine.RemoveEntryFromRaftLog(testCmdIndex), true);
 
-                _ = leader.ServerStore.SendToLeaderAsync(new AddDatabaseCommand(Guid.NewGuid().ToString())
+                Assert.True(res);
+            }
+
+            
+            foreach (var server in Servers)
+            {
+                long lastCommittedIndex = 0;
+                var res = await WaitForValueAsync(() =>
                 {
-                    Record = new DatabaseRecord("Toli")
+                    using (server.ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+                    using (context.OpenReadTransaction())
                     {
-                        Topology = new DatabaseTopology
-                        {
-                            Members = new List<string> { "A", "B", "C" },
-                            Rehabs = new List<string> { },
-                            ReplicationFactor = 3
-                        }
-                    },
-                    Name = "Toli"
-                });
-
-                foreach (var server in Servers)
-                {
-                    Assert.False(server.ServerStore.DatabasesLandlord.IsDatabaseLoaded("Toli"));
-                }
+                        server.ServerStore.Engine.GetLastCommitIndex(context, out lastCommittedIndex, out long _);
+                    }
+                    return lastCommittedIndex > testCmdIndex;
+                }, true);
+                Assert.True(res, $"State machine is stuck. raft index was {testCmdIndex}, after remove raft entry index is {lastCommittedIndex} ");
+            }
 
-                long index = leaderLastIndex;
-                foreach (var server in Servers)
-                {
-                    var res = await WaitForValueAsync(() => server.ServerStore.Engine.RemoveEntryFromRaftLog(index + 1), true);
+            foreach (var server in Servers)
+            {
+                var val = await WaitForValueAsync(() => server.ServerStore.DatabasesLandlord.IsDatabaseLoaded(database), true);
+                Assert.True(val);
+            }
 
-                    Assert.True(res);
-                }
+        }
 
-                long index2 = 0;
-                foreach (var server in Servers)
+        private async Task<long> WaitForCommandIndex(List<RavenServer> nodes, string commandType, long timeout = 15_000, long interval = 100)
```

Comment:
why not to use `WaitForRaftCommandToBeAppliedInClusterAsync` ? it does the same thing, no?

Reply:
Because the command is never applied..

Reply:
so let's create a proper method out of this 
call it `WaitForRaftCommandToBeAppendedInClusterAsync` and move it to be under `ClusterTestBase2`

Reply:
Done